markdown/markdown.go

```diff
@@ -317,6 +329,7 @@ const MetadataURL = "url"
 func (g *gioNodeRenderer) renderLink(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
 	n := node.(*ast.Link)
 	if entering {
+		g.EnsureTextSeparationFromPrevious()
```

Comment:
I'm not sure about this. Why does this need a special case? If there's no space in the source (`hi[there](http://example.com)person`) should we not respect that (rendered as `hithereperson` with `there` clickable)?

Reply:
If there is a newline directly before the link, it will be ignored otherwise and no space will be added. Rendering https://www.gnu.org/licenses/agpl.md will not add a space before the first link without this code. I see your point, but I don't know how to fix in another way :-(.

Reply:
Thinking more about this, one might argue that this pull request improves the situation, but introduces different (less common) issues. Maybe other options should be checked with the upstream markdown library (maybe creating an issue there). For now, I could also offer to close this pull request as it offers no perfect solution.